docs/src/manuals/delta-node.md

```diff
@@ -0,0 +1,133 @@
+
+## Gaussian Case
+
+In the context of Gaussian distributions, we recommend either the `Linearization` or `Unscented` method for delta node approximation. The `Linearization` method provides a first-order approximation, while the `Unscented` method delivers a more precise second-order approximation. It's worth noting that while the `Unscented` method is more accurate, it also requires a little more computational resources and has additional hyper-parameters, which may affect the result. By default, delta node approximation in RxInfer.jl employs the `Unscented` method.
+
+
+For clarity, consider the following example:
+
+```@example delta_node_example
+using RxInfer
+
+@model function delta_node_example()
+    z = datavar(Float64)
+    x ~ Normal(mean=0.0, var=1.0)
+    y ~ tanh(x)
+    z ~ Normal(mean=y, var=1.0)
+end
+```
+
+To perform inference on this model, designate the approximation method for the delta node (here, the `tanh` function) using the `@meta` specification:
+
+```@example delta_node_example
+delta_meta = @meta begin 
+    tanh() -> Linearization()
+end
+```
+or
+```@example delta_node_example
+delta_meta = @meta begin 
+    tanh() -> Unscented()
+end
+```
+
+For a deeper understanding of the [`Unscented`](@ref) method and its parameters, consult the package documentation.
+
+Given the invertibility of `tanh`, indicating its inverse function can optimize inference outcomes:
+
+```@example delta_node_example
+delta_meta = @meta begin 
+    tanh() -> DeltaMeta(method = Linearization(), inverse = atanh)
+end
+```
+
+To execute the inference:
+
+```@example delta_node_example
+inference(model = delta_node_example(), meta=delta_meta, data = (z = 1.0,))
+```
+
+This methodology is consistent even when the delta node is associated with multiple nodes. For instance:
+
+```@example delta_node_example
+f(x, g) = x*tanh(g)
+```
+
+```@example delta_node_example
+@model function delta_node_example()
+    z = datavar(Float64)
+    x ~ Normal(mean=1.0, var=1.0)
+    g ~ Normal(mean=1.0, var=1.0)
+    y ~ f(x, g)
+    z ~ Normal(mean=y, var=0.1)
+end
+```
+
+The corresponding meta specification is:
+
+```@example delta_node_example
+delta_meta = @meta begin 
+    f() -> DeltaMeta(method = Linearization())
+end
+```
+or simply
+```@example delta_node_example
+delta_meta = @meta begin 
+    f() -> Linearization()
+end
+```
+
+If specific functions outline the backward relation of variables within the `f` function, you can provide a tuple of inverse functions in the order of the variables:
+
+```julia
+delta_meta = @meta begin 
+    f() -> DeltaMeta(method = Linearization(), inverse=(f_back_x, f_back_g))
+end
+```
+
+## Exponential Family Case
+
+When the delta node is associated with nodes from the exponential family (excluding Gaussians), the `Linearization` and `Unscented` methods are not applicable. In such cases, the CVI (Conjugate Variational Inference) becomes essential. Here's a modified example:
```

Comment:
You mean when the incoming messages are non-gaussian?

Reply:
Strictly speaking, that is how it should be phrased. However, I haven't introduced the concept of messages here, so I wanted to refrain from it. Practically "associated" or "connected" to exponential family nodes will require you to use CVI or smth like that.